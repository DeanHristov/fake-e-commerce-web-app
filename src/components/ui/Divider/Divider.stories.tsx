import type { Meta, StoryObj } from '@storybook/react';

import Divider, { IDividerProps } from './index';

const meta: Meta<IDividerProps> = {
  title: 'UI/Divider',
  component: Divider,
  argTypes: {},
};

export default meta;

type Story = StoryObj<IDividerProps>;

export const Default: Story = {
  args: {},
};

export const WithGradient: Story = {
  args: {
    gradient: true,
    color: 'via-blue-500',
  },
};
