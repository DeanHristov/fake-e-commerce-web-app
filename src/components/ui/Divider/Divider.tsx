'use client';

import { FC } from 'react';

export interface IDividerProps {
  color?: string;
  abv?: boolean;
  gradient?: boolean;
}

const Divider: FC<IDividerProps> = ({ color, gradient = false }) => {
  const gradientStyles = `w-full h-px m-auto bg-gradient-to-r from-transparent ${
    color ?? 'via-gray-400'
  } to-transparent`;
  const defaultStyles = `w-full m-auto border-b ${color ?? 'border-gray-400'}`;

  return <div className={gradient ? gradientStyles : defaultStyles} />;
};

export default Divider;
