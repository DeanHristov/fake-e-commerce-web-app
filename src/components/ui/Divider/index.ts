import Divider, { IDividerProps } from './Divider';

export type { IDividerProps };

export default Divider;
