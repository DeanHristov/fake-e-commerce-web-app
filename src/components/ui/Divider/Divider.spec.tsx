import { render } from '@testing-library/react';
import Divider from './index';

describe('UI/Divider <Divider {...} />', () => {
  it('Should render default state', () => {
    const { container } = render(<Divider />);

    expect(container).toMatchSnapshot();
  });

  it('Should render with pink gradient', () => {
    const { container } = render(<Divider gradient color="via-pink-600" />);

    expect(container).toMatchSnapshot();
  });
});
