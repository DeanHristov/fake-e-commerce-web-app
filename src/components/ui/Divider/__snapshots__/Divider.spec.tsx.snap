// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Divider <Divider {...} /> Should render default state 1`] = `
<div>
  <div
    class="w-full m-auto border-b border-gray-400"
  />
</div>
`;

exports[`UI/Divider <Divider {...} /> Should render with pink gradient 1`] = `
<div>
  <div
    class="w-full h-px m-auto bg-gradient-to-r from-transparent via-pink-600 to-transparent"
  />
</div>
`;
